### PR TITLE
Updates #404: Add validations for Internet Address.

### PR DESCRIPTION
**Pull request by devyanikota**
Signed-off-by: Devyani Kota <devyanikota@gmail.com>

### CODE REVIEW

File: glusterd2/xlator/options/options.go [hunk 1]
```diff
@@ -186,12 +187,29 @@ func ValidateInt(o *Option, val string) error {
 
 // ValidateInternetAddress validates the Internet Address
 func ValidateInternetAddress(o *Option, val string) error {
-	return ErrInvalidArg
+	_, _, err := net.ParseCIDR(val)
+	if err != nil {
+		return ErrInvalidArg
+	} else if validate.IsIP(val) != true {
```
Comment (atinmu) [inline glusterd2/xlator/options/options.go]: Is `validate.IsIP` taking consideration of validating both Ipv4 and IPv6?

Comment (atinmu) [inline glusterd2/xlator/options/options.go]: Shouldn't validateInternetAddress also consider a valid hostname to be a correct val passed in?

Comment (devyanikota) [reply glusterd2/xlator/options/options.go]: @atinmu yes, validate.IsIP validates IPv4 and IPv6 addresses too.

Comment (devyanikota) [reply glusterd2/xlator/options/options.go]: @atinmu validate.IsHost() takes care of that.

**Comment by devyanikota** [comment]
@kshlm @atinmu while discussing with @aravindavk, we realized that since  GF_OPTION_TYPE_INTERNET_ADDRESS  doesn't seem to have the _SETTABLE_ flag, so we might not need the validations?
Please give your opinion. Thank you.

File: glusterd2/xlator/options/options.go (hunk 1 above)
Comment (atinmu) [reply glusterd2/xlator/options/options.go]: My bad, sorry.

**Comment by atinmu** [comment]
> @kshlm @atinmu while discussing with @aravindavk, we realized that since GF_OPTION_TYPE_INTERNET_ADDRESS doesn't seem to have the SETTABLE flag, so we might not need the validations?

@devyanikota I may not have the complete background of the intention of the PR, but if we don't need the validations then why are these functions in place? Wouldn't a peer probe code path will hit this validation?

**Comment by aravindavk** [comment]
> @devyanikota I may not have the complete background of the intention of the PR, but if we don't need the validations then why are these functions in place? Wouldn't a peer probe code path will hit this validation?

I think we don't need validation if SETTABLE is not specified.

There are some internal options, which will gets added in volfile but user will not set those options. Mostly picked from peer info and added during volfile generation. Even if we add these validations, it will never gets used since none of these options are set by API users.

I don't think peer probe will use this validation. Others can confirm.

**Comment by atinmu** [comment]
> I think we don't need validation if SETTABLE is not specified.There are some internal options, which will gets added in volfile but user will not set those options. Mostly picked from peer info and added during volfile generation. Even if we add these validations, it will never gets used since none of these options are set by API users. I don't think peer probe will use this validation. Others can confirm.

If I can map this functionality back in GD1 & CLI I see `valid_internet_address` function is being called in `cli_cmd_peer_probe_cbk` .

**Comment by kshlm** [comment]
While INTERNET_ADDRESS doesn't have any settable options, we do have settable INTERNET_ADDRESS_LIST (auth.allow/reject) which basically needs the same checks. As @devyanikota has implemented `ValidateInternetAddressList` on top of `ValidateInternetAddress`, I don't see a problem in keeping this as is.

As for the other comments about validating arguments in CLI, they could technically call the validation functions in the options package to perform the validation. The only thing that sort of prevents this right now is that the `options` package is under `glusterd2` directory, but we can move it to be under `pkg`.

**Comment by atinmu** [comment]
So this is more from a option specific validation function than a generic
one then. @devyanikota would you take care of the peer probe validations as
highlighted in a separate PR then?

**Pull request merged**